src/main/java/com/google/sps/servlets/SavedRecipeServlet.java

```diff
@@ -0,0 +1,131 @@
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.json.simple.JSONObject;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+
+/** Servlet that posts and gets saved recipe information in Datastore */
+@WebServlet("/saved-recipe")
+public class SavedRecipeServlet extends HttpServlet {
+  
+  /** Gets SavedRecipe using the recipe's id */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    response.setContentType("application/json");
+    JSONObject responseMap = new JSONObject();
+
+    // get recipeId from request parameter
+    Long recipeId = Long.parseLong(request.getParameter("recipeId"));
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("SavedRecipe")
+      .setFilter(new Query.FilterPredicate("recipeId", Query.FilterOperator.EQUAL, recipeId));
+    PreparedQuery results = datastore.prepare(query);
+    Entity entity = results.asSingleEntity();
+    if (entity == null) {
+      responseMap.put("recipeIsSaved", false);
+    } else {
+      String title = (String) entity.getProperty("title");
+      String imageUrl = (String) entity.getProperty("imageUrl");
+      String sourceUrl = (String) entity.getProperty("sourceUrl");
+      ArrayList<String> dietaryNeedsStrings = (ArrayList<String>) entity.getProperty("dietaryNeeds");
+      
+      // convert string to Diet enum because datastore stores dietaryNeeds as a list of strings
+      ArrayList<SavedRecipe.Diet> dietaryNeeds = new ArrayList<>();
+      for (String dietString: dietaryNeedsStrings) {
+        switch(dietString) {
+          case "VEGETARIAN":
+            dietaryNeeds.add(SavedRecipe.Diet.VEGETARIAN);
+            break;
+          case "VEGAN":
+            dietaryNeeds.add(SavedRecipe.Diet.VEGAN);
+            break;
+          case "GLUTENFREE":
+            dietaryNeeds.add(SavedRecipe.Diet.GLUTENFREE);
+            break;
+          case "DAIRYFREE":
+            dietaryNeeds.add(SavedRecipe.Diet.DAIRYFREE);
+            break;
+          default: break;
+        }
+      }
+      SavedRecipe savedRecipeObject = SavedRecipe.builder().setId(recipeId).setTitle(title).
+        setImage(imageUrl).setSourceUrl(sourceUrl).setDietaryNeeds(dietaryNeeds).build();
+          
+      responseMap.put("savedRecipe", savedRecipeObject);
+      responseMap.put("recipeIsSaved", true);
+    }
+    
+    Gson gson = new Gson();   
+    String json = gson.toJson(responseMap);
```

Comment:
I believe you can do this with org.simple.json by using responseMap.toString(), but I am not exactly sure about what difference that change that would make.

Reply:
That change doesn't give me the json, it gives a syntax error for unexpected token in json.